core/src/main/java/com/digitalpebble/storm/crawler/protocol/http/HttpProtocol.java

```diff
@@ -202,22 +202,23 @@ public Config getConf() {
     }
 
     @Override
-    public ProtocolResponse getProtocolOutput(String urlString, Map<String, String[]> knownMetadata)
-            throws Exception {
+    public ProtocolResponse getProtocolOutput(String urlString,
+            Metadata knownMetadata) throws Exception {
 
         URL u = new URL(urlString);
 
-        if (knownMetadata.containsKey("If-Modified-Since"))
-            this.ifModifiedSince = knownMetadata.get("If-Modified-Since")[0];
+        String ifModifiedSince = knownMetadata
+                .getFirstValue("If-Modified-Since");
+        if (StringUtils.isNotBlank(ifModifiedSince))
+            this.ifModifiedSince = ifModifiedSince;
 
         long startTime = System.currentTimeMillis();
         HttpResponse response = new HttpResponse(this, u); // make a request
-        HashMap<String, String[]> metadata = response.getHeaders();
+        Metadata metadata = response.getHeaders();
 
         if (this.responseTime) {
             int elapsedTime = (int) (System.currentTimeMillis() - startTime);
-            metadata.put("_rst_",
-                    new String[] { Integer.toString(elapsedTime) });
+            metadata.setValue("_rst_", Integer.toString(elapsedTime));
```

Comment:
that's how we inherited it from Nutch, can think about that one in a separate issue maybe